code/game/objects/items/books/skill_book.dm

```diff
@@ -82,64 +82,106 @@ Skill books that increase your skills while you activate and hold them
 /datum/skill_buff/skill_book
 	limit = 1 // you can only read one book at a time nerd, therefore you can only get one buff at a time
 
-/obj/item/book/skill/attack_self(mob/user)
+/obj/item/book/skill/proc/check_can_read(mob/user)
 	if(!skill || (custom && progress == SKILLBOOK_PROG_NONE))
 		to_chat(user, SPAN_WARNING("The textbook is blank!"))
-		return
+		return FALSE
 	if(custom && progress < SKILLBOOK_PROG_FINISH)
 		to_chat(user, SPAN_WARNING("The textbook is unfinished! You can't learn from it in this state!"))
-		return
+		return FALSE
 	if(!ez_read &&!user.skill_check(SKILL_LITERACY, SKILL_BASIC))
 		to_chat(user, SPAN_WARNING(pick("Haha, you know you can't read. Good joke. Put [title] back.","You open up [title], but there aren't any pictures, so you close it again.","You don't know how to read! What good is this [name] to you?!")))
-		return
+		return FALSE
 
-	if(reading) //Close book, get rid of buffs
-		src.unlearn(user)
-		to_chat(user, SPAN_NOTICE("You close the [name]. That's enough learning for now."))
-		reading = FALSE
-		return
+	if(reading)
+		if(reading != weakref(user))
+			to_chat(user, SPAN_WARNING("\The [reading.resolve()] is already using this textbook!"))
+		else
+			to_chat(user, SPAN_WARNING("You are already using this textbook!"))
+		return FALSE
 
 	if(user.too_many_buffs(/datum/skill_buff/skill_book))
 		to_chat(user, SPAN_WARNING("You can't read two books at once!"))
-		return
+		return FALSE
 
 	if(!user.skill_check(skill, skill_req))
 		to_chat(user, SPAN_WARNING("[title] is too advanced for you! Try something easier, perhaps the \"For Idiots\" edition?"))
-		return
+		return FALSE
+
 	if(user.get_skill_value(skill) > skill_req)
 		to_chat(user, SPAN_WARNING("You already know everything [title] has to teach you!"))
-		return
+		return FALSE
+
+	return TRUE
+
+/obj/item/book/skill/attack_self(mob/user)
+
+	if(reading && reading == weakref(user))
```

Comment:
I believe the correct check is always `user && user == reading?.resolve()` or equivalent. Avoids creating unnecessary weakrefs and is independent of the weakref implementation.